### DISCUSSION TITLE
error-stack: Move `error-stack` to its own repository?

### DISCUSSION THREAD

**Discussion by JonasCir** [hashintel/hash#686] [error-stack (Help)] [answered]
Hey there,
I was reading the last issue of [This Week in Rust](https://this-week-in-rust.org/blog/2022/06/15/this-week-in-rust-447/) where your blog post on releasing `error-stack` was featured.

I really like the approach you are taking, great work :tada: ! I think this has the potential to set a new standard for the quality/usability of error reporting in general, not only for Rust. I currently work with Java EE where 90% of your stack trace is irrelevant, so it is refreshing to see that other ecosystems focus on providing clear and _actionable_ error messages.

While I was following the link to the source code, I noticed that it is included in this repository together with other projects. I think it might be beneficial to move this to a dedicated repository. This would be less confusing, issue reports would not mix up, and it likely helps to give the project more visibility.

Thank you for your work!

**Comment by vilkinsons** [comment, marked as answer]
Thanks for the kind words, Jonas.

While we have no plans to move the `error-stack` lib out of our public monorepo at this time, maintainability and QoL are things are care a lot about, and we'll keep an eye on this.

Our current approach to separating concerns on GitHub (within a single repo) primarily leverages labeling and categories, and appears to be working well right now. D

**Comment by vilkinsons** [comment]
Unlocking this thread so there is one central place to discuss this issue, but our position remains unchanged from the above.
